### ISSUE TITLE
Uncaught TypeError: Cannot read property 'show' of undefined

### ISSUE THREAD

**Issue by smddzcy** [thomasjo/atom-latex#268]
Whenever I press "Command + W" (or let's say, whenever I try to close a file) it gives me an error and prevents me from closing anything. 

Here is the error:
`Uncaught TypeError: Cannot read property 'show' of undefined`

Here is where it happens:
`/Users/smddzcy/.atom/packages/latex/lib/loggers/default-logger.js:40`

And the stack trace:

```
TypeError: Cannot read property 'show' of undefined
    at DefaultLogger.updateStatus (/Users/smddzcy/.atom/packages/latex/lib/loggers/default-logger.js:40:17)
    at DefaultLogger.hide (/Users/smddzcy/.atom/packages/latex/lib/loggers/default-logger.js:74:10)
    at Object.hide (/Users/smddzcy/.atom/packages/latex/lib/latex.js:149:24)
    at atom-workspace.coreClose (/Users/smddzcy/.atom/packages/latex/lib/main.js:32:26)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:114:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

I haven't looked at the whole code but adding a `if(latex.status.hasOwnProperty('show'))` check fixed this for now. 

If you need any additional information, please let me know. Thanks for maintaining this great package!

**Comment by yitzchak** [comment]
@smddzcy Do you have the status bar package enabled and do you have a `LaTeX Log` indicator in your status bar after a build?

**Comment by smddzcy** [comment]
@yitzchak Status bar package is enabled but there is no `Latex Log` indicator, there is a spinning `Latex Rebuild`.

**Comment by yitzchak** [comment]
Does the `LaTeX Rebuild` stay spinning even when the build is done?

**Comment by smddzcy** [comment]
Yes.

**Comment by yitzchak** [comment]
What version of `atom-latex` do you have?

**Comment by smddzcy** [comment]
It is `0.38.1`.

**Comment by yitzchak** [comment]
Can you try toggling the developer pane and see if there are any error messages in the console aside from the trace you reported?

**Comment by nschneid** [comment]
Encountering this with 0.38.1 as well. I have a window where "LaTeX Build" is spinning (the build is apparently hanging). For any file in that window—including .html files etc.—when I do a find in the file, and then hit Esc to close the search pane, it triggers the error.

**Comment by nschneid** [comment]
In the developer console, there are multiple errors logged each time this happens:


/Users/nathan/.atom/packages/latex/lib/loggers/default-logger.js:78 Uncaught TypeError: Cannot read property 'show' of undefinedmodule.exports.KeymapManager.dispatchCommandEvent @ /Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580

https://is.gd/create.php?format=simple Failed to load resource: the server responded with a status of 400 ()

/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:3256 Uncaught (in promise)

**Comment by yitzchak** [comment]
@nschneid and @smddzcy: Do you have a sample document that this happens with? Also what are all your settings for the latex package? Lastly, please try `latexmk -v` from the command line to do a version check on latexmk.

**Comment by nschneid** [comment]
`Latexmk, John Collins, 22 April 2016. Version 4.45`. Don't have time to get the other info right now....

**Comment by yitzchak** [comment]
@smddzcy and @nschneid What OS are you on?

**Comment by nschneid** [comment]
I'm on Mac OS 10.11.6 El Capitan

**Comment by yitzchak** [comment]
@nschneid What PDF viewer are you using?

**Comment by nschneid** [comment]
> @nschneid What PDF viewer are you using?

Preview. I haven't set any of the viewer paths in the settings.

**Comment by yitzchak** [comment]
Do you have any other LaTeX packages installed?

**Comment by nschneid** [comment]
> Do you have any other LaTeX packages installed?

In Atom, just latex and language-latex.

**Comment by yitzchak** [comment]
@nschneid and @smddzcy: There have been a few changes in v0.39.0 that might affect your issue. We have not been able reproduce the issue yet, but please upgrade on the off chance that it has been resolved by some of recent changes.

**Comment by yitzchak** [comment]
@nschneid and @smddzcy: We just published v0.40.0. Please let us know if the issues still exist.

**Comment by yitzchak** [comment]
I presuming this is fixed due to lack of confirmation. Will reopen if needed.
